strawberryfields/ops.py

```diff
@@ -1745,12 +1745,9 @@ def __init__(self, S, vacuum=False, tol=1e-10):
             X2 = O2[:N, :N]
             P2 = O2[N:, :N]
 
-            self.U1 = X1+1j*P1
-            self.U2 = X2+1j*P2
-            self.Sq = np.diagonal(smat)[:N]
-
-        self.ns = N
```

Comment:
If I recall, setting `ns` here allowed a exception to be raised if `2*len(modes) != len(S)`, but I could be wrong?

Reply:
Fixed.